### ISSUE TITLE
Question about  Epoch and EarlyStoping?

### ISSUE THREAD

**Issue by AlongWY** [jcyk/AMR-gs#3]
# Question about Epoch Setting and EarlyStoping
Hi Deng,

First congratulations on your ACL work.

I read your paper and code and then i notice that your paper mentioned about **EarlyStopping** but i don't find it in the code. Should I stop it by hand? Another question is about the **Epoch Setting**, the 
epoch setting in your shell code is **100000**, it's so big and the train time will very very long. Is there some extra detail i didn't noticed? 

Thanks a lot.

Kind regards,
Feng Yunlong

**Comment by jcyk** [comment]
Hi,

Actually,  the epoch setting is never used. The Earlystopping is conducted in the following way:

1. Monitor the training process and stop training once the number of maximum training steps is reached.
2. The training process will produce many checkpoints and the corresponding outputs on dev set
3. Select the best checkpoint by evaluating the dev output files (need to do postprocessing first).
4. Test the best checkpoint on test set and report the results.

Sorry for the confusion. You may wonder why we did not include step3 in the training code. The reason is that we did not feel convenient to invoke postprocessing in python code and computing Smatch is very slow (FastSmatch is recommended).

Deng

**Comment by AlongWY** [comment]
Thanks for your help @jcyk
